### ISSUE TITLE
Just upgraded to 1.0.0-rc3

### ISSUE THREAD

**Issue by aphillipo** [rackt/react-router#2217]
A few notes:

1. The upgrade guide was superb!  Thanks so much for that.  Made it 100% easier than I was expecting.
2. The History mixin does not seem to work properly on 1.0.0-rc3 and installing from github has fixed my issues.
3. Maybe this means it's time for RC4?

**Comment by knowbody** [comment]
RC3 was released yesterday lol

the `History` is now in the separate repo. Please see the docs :)

**Comment by chentsulin** [comment]
@knowbody what he says is History `mixin`: https://github.com/rackt/react-router/blob/master/modules/History.js, not history module.

**Comment by aphillipo** [comment]
Well I'm just giving feedback that RC3 does not work with the History mixin.  I'd suggest rereleasing RC3 seeing as it doesn't work right and you have obviously fixed something in the last 24 hours which fixes the History mixin.  Even if you think opening this bug was stupid it'll probably help someone who is having issues with said mixin so I'm not that bothered...

**Comment by mjackson** [comment]
What do you mean by "doesn't work"? This is news to me. Can you make a
failing test?

**Comment by aphillipo** [comment]
Um... I'll have a think about how to do it... I have a large codebase that I just converted from one of the very early betas with Navigation and TransitionHook to History and Lifecycle respectively.

Actually, hang on - maybe it was a diff between github version and 1.0.0-rc2... Let me just check that npm rc3 100% fails.

**Comment by aphillipo** [comment]
Weird, downgraded to what's on npm 1.0.0-rc3 and I get a different error than before:

Uncaught TypeError: Cannot read property 'search' of undefined 

But it works with github version; I'm going to stick with that for now as it works *perfectly* :+1:

**Comment by mjackson** [comment]
@aphillipo :+1:

**Comment by jquense** [comment]
I am also getting `Uncaught TypeError: Cannot read property 'search' of undefined` when I switch certain routes. I can't figure out what is consistent about it tho...

in computeChangedRoutes, nextState.locaton is undefined.

```js
function routeQueryChanged(prevState, nextState) {
  return prevState.location.search !== nextState.location.search; 
}
```

**Comment by taion** [comment]
Do you have a stack trace?

**Comment by jquense** [comment]
here ya go, not pretty

```js
routeQueryChanged (computeChangedRoutes.js:18)
(anonymous function) (computeChangedRoutes.js:41)
computeChangedRoutes (computeChangedRoutes.js:40)
(anonymous function) (useRoutes.js:156)
done (AsyncUtils.js:13)
(anonymous function) (matchRoutes.js:131)
(anonymous function) (matchRoutes.js:100)
done (AsyncUtils.js:13)
(anonymous function) (matchRoutes.js:131)
(anonymous function) (matchRoutes.js:100)
done (AsyncUtils.js:13)
(anonymous function) (matchRoutes.js:131)
(anonymous function) (matchRoutes.js:81)
getIndexRoute (matchRoutes.js:31)
(anonymous function) (matchRoutes.js:75)
matchRouteDeep (matchRoutes.js:84)
(anonymous function) (matchRoutes.js:129)
next (AsyncUtils.js:20)
loopAsync (AsyncUtils.js:26)
matchRoutes (matchRoutes.js:128)
(anonymous function) (matchRoutes.js:94)
getChildRoutes (matchRoutes.js:13)
matchRouteDeep (matchRoutes.js:89)
(anonymous function) (matchRoutes.js:129)
next (AsyncUtils.js:20)
(anonymous function) (matchRoutes.js:133)
(anonymous function) (matchRoutes.js:102)
done (AsyncUtils.js:13)
next (AsyncUtils.js:22)
(anonymous function) (matchRoutes.js:133)
matchRouteDeep (matchRoutes.js:110)
(anonymous function) (matchRoutes.js:129)
next (AsyncUtils.js:20)
(anonymous function) (matchRoutes.js:133)
matchRouteDeep (matchRoutes.js:110)
(anonymous function) (matchRoutes.js:129)
next (AsyncUtils.js:20)
(anonymous function) (matchRoutes.js:133)
matchRouteDeep (matchRoutes.js:110)
(anonymous function) (matchRoutes.js:129)
next (AsyncUtils.js:20)
loopAsync (AsyncUtils.js:26)
matchRoutes (matchRoutes.js:128)
(anonymous function) (matchRoutes.js:94)
getChildRoutes (matchRoutes.js:13)
matchRouteDeep (matchRoutes.js:89)
(anonymous function) (matchRoutes.js:129)
next (AsyncUtils.js:20)
loopAsync (AsyncUtils.js:26)
matchRoutes (matchRoutes.js:128)
(anonymous function) (matchRoutes.js:94)
getChildRoutes (matchRoutes.js:13)
matchRouteDeep (matchRoutes.js:89)
(anonymous function) (matchRoutes.js:129)
next (AsyncUtils.js:20)
(anonymous function) (matchRoutes.js:133)
(anonymous function) (matchRoutes.js:102)
done (AsyncUtils.js:13)
next (AsyncUtils.js:22)
(anonymous function) (matchRoutes.js:133)
matchRouteDeep (matchRoutes.js:110)
(anonymous function) (matchRoutes.js:129)
next (AsyncUtils.js:20)
loopAsync (AsyncUtils.js:26)
matchRoutes (matchRoutes.js:128)
(anonymous function) (matchRoutes.js:94)
getChildRoutes (matchRoutes.js:13)
matchRouteDeep (matchRoutes.js:89)
(anonymous function) (matchRoutes.js:129)
next (AsyncUtils.js:20)
loopAsync (AsyncUtils.js:26)
matchRoutes (matchRoutes.js:128)
transitionHook (useRoutes.js:143)
runTransitionHook (runTransitionHook.js:12)
(anonymous function) (useQueries.js:61)
runTransitionHook (runTransitionHook.js:12)
(anonymous function) (createHistory.js:114)
next (AsyncUtils.js:19)
loopAsync (AsyncUtils.js:25)
confirmTransitionTo (createHistory.js:113)
transitionTo (createHistory.js:139)
pushState (createHistory.js:155)
pushState (useQueries.js:73)
```

**Comment by mjackson** [comment]
@jquense Your problem was fixed in 6f8ceac463f49f96bec7d429071c5bd3746c1e7c. We'll release an rc4 this week with this fix.
